hazelcast-client/src/main/java/com/hazelcast/client/config/QueryCacheConfigBuilderHelper.java

```diff
@@ -16,152 +16,8 @@
-        final Node maxSizePolicy = node.getAttributes().getNamedItem("max-size-policy");
-        final Node evictionPolicy = node.getAttributes().getNamedItem("eviction-policy");
-        if (size != null) {
-            evictionConfig.setSize(Integer.parseInt(getTextContent(size)));
-        }
-        if (maxSizePolicy != null) {
-            evictionConfig.setMaximumSizePolicy(
-                    EvictionConfig.MaxSizePolicy.valueOf(
-                            upperCaseInternal(getTextContent(maxSizePolicy)))
-            );
-        }
-        if (evictionPolicy != null) {
-            evictionConfig.setEvictionPolicy(
-                    EvictionPolicy.valueOf(
-                            upperCaseInternal(getTextContent(evictionPolicy)))
-            );
-        }
-        return evictionConfig;
-    }
-
-    private void handleEntryListeners(QueryCacheConfig queryCacheConfig, Node childNode) {
-        for (Node listenerNode : childElements(childNode)) {
-            if ("entry-listener".equals(cleanNodeName(listenerNode))) {
-                NamedNodeMap listenerNodeAttributes = listenerNode.getAttributes();
-                boolean incValue
-                        = getBooleanValue(getTextContent(listenerNodeAttributes.getNamedItem("include-value")));
-                boolean local
-                        = getBooleanValue(getTextContent(listenerNodeAttributes.getNamedItem("local")));
-                String listenerClass = getTextContent(listenerNode);
-                queryCacheConfig.addEntryListenerConfig(
-                        new EntryListenerConfig(listenerClass, local, incValue));
-            }
-        }
-    }
-
-    private void queryCachePredicateHandler(Node childNode, QueryCacheConfig queryCacheConfig) {
-        NamedNodeMap predicateAttributes = childNode.getAttributes();
-        String predicateType = getTextContent(predicateAttributes.getNamedItem("type"));
-        String textContent = getTextContent(childNode);
-        PredicateConfig predicateConfig = new PredicateConfig();
-        if ("class-name".equals(predicateType)) {
-            predicateConfig.setClassName(textContent);
-        } else if ("sql".equals(predicateType)) {
-            predicateConfig.setSql(textContent);
-        }
-        queryCacheConfig.setPredicateConfig(predicateConfig);
-    }
-
-    private void queryCacheIndexesHandle(Node n, QueryCacheConfig queryCacheConfig) {
-        for (Node indexNode : childElements(n)) {
-            if ("index".equals(cleanNodeName(indexNode))) {
-                final NamedNodeMap attrs = indexNode.getAttributes();
-                boolean ordered = getBooleanValue(getTextContent(attrs.getNamedItem("ordered")));
-                String attribute = getTextContent(indexNode);
-                queryCacheConfig.addIndexConfig(new MapIndexConfig(attribute, ordered));
-            }
-        }
-    }
-
+interface QueryCacheConfigBuilderHelper {
```

Comment:
Minor: Could you add Javadoc to explain the purpose of this interface (and therefore the reason for the whole classes hierarchy)?

Reply:
Done.